graphql_compiler/compiler/ir_lowering_match/__init__.py

```diff
@@ -103,4 +103,9 @@ def lower_ir(ir_blocks, location_types, type_equivalence_hints=None):
     compound_match_query = lower_context_field_expressions(
         compound_match_query)
 
-    return compound_match_query
+    lowered_match_queries = []
+    for match_query in compound_match_query.match_queries:
+        new_match_query = truncate_repeated_single_step_traversals(match_query)
+        lowered_match_queries.append(new_match_query)
+
+    return CompoundMatchQuery(match_queries=lowered_match_queries)
```

Comment:
Can you extract this into a function? I would like to keep as little logic as possible in this function itself, and just call out to other functions that implement the actual lowering process.